Infra / 7833 Remove Enhanced Measurement Feature Flag

10upsimon opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #7833

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

zutigrm commented on assets/js/modules/analytics/components/setup/SetupForm.test.js:
```diff
@@ -279,11 +314,6 @@ describe( 'SetupForm', () => {
-			status: 200,
-			body: [],
-		} );
-
```
removed as not needed, since we already populate the state with `webDataStreamsBatch` from fixtures.

zutigrm commented on assets/js/modules/analytics/components/setup/SetupForm.test.js:
```diff
@@ -315,10 +345,6 @@ describe( 'SetupForm', () => {
-		expect( fetchMock ).toHaveFetchedTimes(
-			1,
-			getWebDataStreamsBatchRegexp
-		);
```
Since `webDataStreamsBatch` is already populated in state, this fetch will not happen

techanvil commented on assets/js/modules/analytics-4/components/dashboard/EnhancedMeasurementActivationBanner/index.test.js:
```diff
@@ -106,17 +108,38 @@ describe( 'EnhancedMeasurementActivationBanner', () => {
 
 		await waitForRegistry();
 
+		// Enhanced measurement settings should update when Enhanced measurement
+		// is enabled via the  "Enable now" CTA.
```
Minor typos:

```suggestion
		// Enhanced measurement settings should update when enhanced measurement
		// is enabled via the "Enable now" CTA.
```

techanvil commented on assets/js/modules/analytics/components/setup/SetupForm.test.js:
```diff
@@ -302,7 +332,7 @@ describe( 'SetupForm', () => {
 		// An additional wait is required in order for all resolvers to finish.
 		await act( async () => {
-			await waitForDefaultTimeouts();
+			await waitForTimeouts( 50 );
```
Very minor, but this looks like an arbitrary value so we might as well use a timeout of `30` here to be consistent with the other places we're using `waitForTimeouts()`, unless `50` is explicitly needed here.

```suggestion
			await waitForTimeouts( 30 );
```

techanvil reviewed (commented):
Nice work here, @10upsimon and @zutigrm. I've left a couple of super trivial comments.

Additionally:
  - For QA testing, we'll need to perform a regression test on the "Enable enhanced measurement" banner as well as the EM Settings. Please can you update the QAB to include a point to cover this.
  - There are a couple of stories that are now redundant and can be removed (see below).



![image](https://github.com/google/site-kit-wp/assets/18395600/d123d9f4-93c0-4a27-9140-3fe85de3deb4)

10upsimon commented:
@techanvil I have addressed your points, and committed your suggestions too, thank you. I've updated the QAB to include:

- Perform regression testing against the "Enable enhanced measurement" banner.
- Perform regression testing against the Enhancement Measurement Settings.

Moving this back to you for a re-review, thank you.

techanvil commented:
> @techanvil I have addressed your points, and committed your suggestions too, thank you. I've updated the QAB to include:
> 
> * Perform regression testing against the "Enable enhanced measurement" banner.
> * Perform regression testing against the Enhancement Measurement Settings.
> 
> Moving this back to you for a re-review, thank you.

Thanks @10upsimon!

To clarify the QAB point, though - what I meant was that you had _already_ provided the QAB for EM Settings regression and needed to include the banner as well.

Noting you're a bit under the weather, to move this along I've addressed the last couple of points myself. I've modified the QAB - I've simplified it a bit, figured it would get a bit on the verbose side to include such a detailed account of the banner as well. I've also removed the VRT reference images for the deleted stories.

techanvil reviewed (approved):
LGTM, nice one @10upsimon and @zutigrm!

Moving this to MR with Evan as per this point in the AC:

> **Before merging a PR for this, please confirm with @aaemnnosttv**.

techanvil reviewed (changes requested):
Hey @10upsimon, this has developed some merge conflicts with the VRT images, please can you resolve them?

techanvil reviewed (approved):
LGTM, nice one @10upsimon!
